src/main/java/frc/robot/statemachine/RobotCommander.java

```diff
@@ -274,39 +292,31 @@ public Command autoScore() {
 		);
 	}
 
-	public Command autoScoreForAutonomous() {
-		Supplier<Command> fullySuperstructureScore = () -> new SequentialCommandGroup(
+	public Command autoScoreForAutonomous(PathPlannerPath path) {
+		Command fullySuperstructureScore = new SequentialCommandGroup(
 			superstructure.elevatorOpening(),
 			superstructure.armPreScore().until(this::isReadyToOpenSuperstructure),
 			superstructure.preScore().until(superstructure::isPreScoreReady),
 			superstructure.scoreWithoutRelease().until(this::isReadyToScore),
 			superstructure.scoreWithRelease()
 		);
 
-		Supplier<Command> driveToPath = () -> swerve.getCommandsBuilder()
-			.driveToPath(
-				() -> robot.getPoseEstimator().getEstimatedPose(),
-				ScoringPathsHelper.getPathByBranch(ScoringHelpers.getTargetBranch()),
-				ScoringHelpers
-					.getRobotBranchScoringPose(ScoringHelpers.getTargetBranch(), StateMachineConstants.ROBOT_SCORING_DISTANCE_FROM_REEF_METERS)
-			);
-
-		return asSubsystemCommand(
-			new DeferredCommand(
-				() -> new ParallelDeadlineGroup(fullySuperstructureScore.get(), driveToPath.get()),
-				Set.of(
-					this,
-					superstructure,
-					swerve,
-					robot.getElevator(),
-					robot.getArm(),
-					robot.getEndEffector(),
-					robot.getLifter(),
-					robot.getSolenoid()
-				)
-			),
-			RobotState.SCORE
+		Command driveToPath = swerve.asSubsystemCommand(
+			PathFollowingCommandsBuilder.followPath(path)
+				.andThen(
+					swerve.getCommandsBuilder()
+						.moveToPoseByPID(
+							() -> robot.getPoseEstimator().getEstimatedPose(),
+							ScoringHelpers.getRobotBranchScoringPose(
+								ScoringHelpers.getTargetBranch(),
+								StateMachineConstants.ROBOT_SCORING_DISTANCE_FROM_REEF_METERS
+							)
```

Comment:
```suggestion
							PathPlannerUtils.getLastPathPose(path)
```
so you wont need to calculate the point again and waste cycle time